addon/resolve-attribute-util.js

```diff
@@ -95,11 +95,13 @@ export function resolveValue(key, value, modelName, store, schema, record, paren
     return resolveReferenceOrReferences(store, record, key, value, reference);
   }
 
-  if (Array.isArray(value)) {
-    return resolveArray(key, value, modelName, store, schema, record, recordData);
-  }
   let nested = computeNestedModel(key, value, modelName, schemaInterface, schema);
   if (nested) {
+    // handle the case where nested returns an array.
+    if (Array.isArray(nested)) {
+      return resolveArray(key, value, modelName, store, schema, record, recordData);
```

Comment:
Also we need tests for this case, where `computeNestedModel` returns an array and app code is able to resolve it correctly

Reply:
added